### ISSUE TITLE
Bringing down PEER0 on a 4 peer network breaks consensus in latest build 

### ISSUE THREAD

**Issue by 2016Nishi** [hyperledger-archives/fabric#1873]
## Description

When PEER0 was stopped on a 4 peer network no consensus happened
## Describe How to Reproduce
1. Start a 4  peer network
2. Deploy chaincode example02 on a peer.
3. Query for A and B on a peer
4. Stop PEER0
5. Send 5 Invokes on PEER1
   Sleep for 2 min
6. Query for A and B values on PEER1, PEER2, PEER3
7. Get chain height from PEER1, PEER2 and PEER3
## Actual Result:

Query values returned on step6 are exact value as in step3
Height remains at 2
## Expected Result:

Query values returned on step6 to take into account 5 invokes that were submitted as in step3
Height to return 7.

Tested could not be verified on commit levels: f8ffcd1 & 04f076b
P.S. 
Test passed on commit level a94f5a7 
[PEERLOGS.zip](https://github.com/hyperledger/fabric/files/318925/PEERLOGS.zip)

**Comment by tuand27613** [comment]
Hi @2016Nishi  ... could you generate debug logs ? thanks !

**Comment by 2016Nishi** [comment]
Please find attached peer debug logs.

[PEER_DEBUGLOGS.zip](https://github.com/hyperledger/fabric/files/319194/PEER_DEBUGLOGS.zip)

**Comment by tuand27613** [comment]
I think this behave test reproduces @2016Nishi 's scenario
[1873.zip](https://github.com/hyperledger/fabric/files/319207/1873.zip)

```
#@doNotDecompose
#    @wip
       Scenario Outline: 4 peers and 1 membersrvc, consensus works if vp0 is stopped TTT3
            Given we compose "<ComposeFile>"
            And I use the following credentials for querying peers:
               | peer |   username  |    secret    |
               | vp0  |  test_user0 | MS9qrN8hFjlE |
               | vp1  |  test_user1 | jGlNl6ImkuDo |
               | vp2  |  test_user2 | zMflqOKezFiA |
               | vp3  |  test_user3 | vWdLCE00vJy0 |
            And I register with CA supplying username "test_user0" and secret "MS9qrN8hFjlE" on peers:
               | vp0 |

            When requesting "/chain" from "vp0"
            Then I should get a JSON response with "height" = "1"

            # Deploy
            When I deploy chaincode "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example02" with ctor "init" to "vp0"
               | arg1 |  arg2 | arg3 | arg4 |
               |  a   |  100  |  b   |  200 |
            Then I should have received a chaincode name
            Then I wait up to "<WaitTime>" seconds for transaction to be committed to peers:
               | vp0  | vp1 | vp2 | vp3 |

            When requesting "/chain" from "vp0"
            Then I should get a JSON response with "height" = "2"

            # STOP vp0
            Given I stop peers:
               | vp0  |
            And I wait "5" seconds

            And I register with CA supplying username "test_user1" and secret "jGlNl6ImkuDo" on peers:
               | vp1 |

            When I invoke chaincode "example2" function name "invoke" on "vp1" "5" times
               |arg1|arg2|arg3|
               | a  | b  | 1 |
            Then I should have received a transactionID
            #Then I wait up to "120" seconds for transaction to be committed to peers:
            #      | vp0  | vp1 | vp2 | vp3 |
            And I wait "120" seconds
            When I query chaincode "example2" function name "query" with value "a" on peers:
               | vp1 | vp2 | vp3 |
            Then I should get a JSON response from peers with "OK" = "95"
               | vp1 | vp2 | vp3 |
            Examples: Consensus Options
               |          ComposeFile                       |   WaitTime   |
              #    |   docker-compose-4-consensus-classic.yml   |      60      |
               |   docker-compose-4-consensus-batch.yml     |      60      |
              #    |   docker-compose-4-consensus-sieve.yml     |      60
```

**Comment by jyellick** [comment]
@tuand27613 Ran your behave test locally against my branch used for #1877 and it passes.

**Comment by tuand27613** [comment]
If you can't wait for PR #1877 to get merged ,

With CORE_PBFT_TIMEOUT_NULLREQUEST=1s , the behave test passes.

**Comment by corecode** [comment]
Shouldn't some timeout kick in and trigger view change?

**Comment by corecode** [comment]
Somehow vp1 never sends any message after vp0 is stopped?

**Comment by corecode** [comment]
The problem is that outstanding requests don't restart the timer.  This is fixed in #1877

**Comment by scottz64** [comment]
The previously failing tests are passing (along with some new tests) using today(Saturday)'s 9:30am images with commit (35326c2), so this issue is fixed by 1877.

**Comment by 2016Nishi** [comment]
Thanks Scott!
Appreciate you taking time over weekend to verify fixes.

**Comment by 2016Nishi** [comment]
Testing 1873 by bringing down one peer (in a 4 peer network) each time, in a continuous loop and validate results match in the remaining 3 peers each time.

Using Commit Level: 35326c2

**Comment by 2016Nishi** [comment]
Issue reported in here has been verified fixed in Commit Level: 35326c2
